src/conn/conn_layered.c

```diff
@@ -1091,6 +1091,94 @@ __wt_conn_is_disagg(WT_SESSION_IMPL *session)
     return (disagg->page_log_meta != NULL);
 }
 
+/*
+ * __disagg_delete_or_fail --
+ *     Delete a file or fail depending on the flag.
+ */
+static int
+__disagg_delete_or_fail(WT_SESSION_IMPL *session, const char *fname, bool fail)
+{
+    bool file_exists;
+
+    WT_RET(__wt_fs_exist(session, fname, &file_exists));
+
+    if (!file_exists)
+        return (0); /* Nothing to do, file does not exist. */
+
+    if (fail)
+        WT_RET_MSG(session, EEXIST,
+          "Disaggregated storage requires a clean directory, but found WiredTiger file %s: "
+          "use 'disaggregated.local_files_action=delete' to remove it.",
+          fname);
+
+    WT_RET(__wt_fs_remove(session, fname, false, false));
+
+    return (0);
+}
+
+/*
+ * __wti_disagg_check_local_files --
+ *     Check for local files that need to be removed before starting in disaggregated mode.
+ *
+ * Disaggregated storage needs to start with a clean directory, for now wipe out the directory if
+ *     starting in disaggregated storage mode. Eventually this should not be necessary but at the
+ *     moment WiredTiger will generate local files in disaggregated storage mode, and MongoDB
+ *     expects to be able to restart without files being present.
+ */
+int
+__wti_disagg_check_local_files(WT_SESSION_IMPL *session, const char *cfg[])
+{
+    WT_CONFIG_ITEM cval;
+    WT_DECL_RET;
+
+    /*
+     * FIXME-WT-14721: As it stands, __wt_conn_is_disagg only works after we have metadata access,
+     * which depends on having run recovery, so the config hack is the simplest way to break that
+     * dependency.
+     */
+    WT_RET(__wt_config_gets(session, cfg, "disaggregated.lose_all_my_data", &cval));
+    if (cval.val == 0)
+        return (0);
+
+    /*
+     * Possible actions for local files are: fail, delete, ignore.
+     *
+     * A reasonable default for Disagg would be to delete all local WT-related files, since they can
+     * be in an inconsistent state anyway. Since this only works together with the
+     * "lose_all_my_data" option, it's considered to be safe enough to be triggered by accident.
+     */
+    bool fail;
+    WT_RET(__wt_config_gets(session, cfg, "disaggregated.local_files_action", &cval));
+    if (WT_CONFIG_LIT_MATCH("fail", cval))
+        fail = true;
+    else if (WT_CONFIG_LIT_MATCH("ignore", cval))
+        return (0);
+    else
+        fail = false; /* Default: delete */
+
+    /*
+     * Avoid using a blanket wildcard like "WiredTiger*" as some files are ok to have, e.g.
+     * WiredTiger.lock.
```

Comment:
I'm not going to argue about the choice here, since the solution is stopgap, but I'd prefer if all WiredTiger files are cleared about. It will make our testing more representative of what will happen when a node is started based only on content in shared storage.